src/shogun/features/Features.cpp

```diff
@@ -307,6 +312,34 @@ void CFeatures::unset_property(EFeatureProperty p)
 	properties &= (properties | p) ^ p;
 }
 
+void CFeatures::eval()
+{
+	if (get_num_preprocessors() > 0 || m_subset_stack->has_subsets() > 0)
+	{
+		eval_features();
+		m_subset_stack->remove_all_subsets();
+		clean_preprocessors();
+	}
+}
+
+Some<CFeatures> CFeatures::view(const SGVector<index_t>& subset)
+{
+	auto feats_view = wrap(this->duplicate());
+	feats_view->add_subset(subset.clone());
+	return feats_view;
+}
+
+Some<CFeatures> CFeatures::view(const std::vector<index_t>& subset)
+{
+	auto feats_view = wrap(this->duplicate());
+
+	auto sg_subset = SGVector<index_t>(subset.size());
+	std::copy(subset.cbegin(), subset.cend(), sg_subset.data());
```

Comment:
`sg_memcpy` ? as both std::vector and SGVector backed by a continuous memory chunk